src/test/App.test.js

```diff
@@ -0,0 +1,51 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import { render, screen } from '@testing-library/react';
+import { MemoryRouter, BrowserRouter } from 'react-router-dom';
+import renderer from 'react-test-renderer';
```

Comment:
- [ ] While you have done a good job, I realized this dependency is not in your package.json file and as a result, it makes 5 of your tests fail.
### Failed tests
![image](https://github.com/Jawad-Nazari/Math-Magicians/assets/86200356/f3815b13-2f7b-4b17-8fb5-d2bcf420149d)

But upon installing it, all the tests were passed. I would implore you to add it, it might be the case that you did not push the file. Kindly fix it.
![image](https://github.com/Jawad-Nazari/Math-Magicians/assets/86200356/9cfeadc1-b33e-41dc-8be8-a5390bfade6f)